src/OpenIddict.EntityFramework/Stores/OpenIddictApplicationStore.cs

```diff
@@ -651,7 +651,32 @@ public virtual ValueTask<ImmutableArray<string>> GetRedirectUrisAsync([NotNull]
         /// whose result returns the instantiated application, that can be persisted in the database.
         /// </returns>
         public virtual ValueTask<TApplication> InstantiateAsync(CancellationToken cancellationToken)
-            => new ValueTask<TApplication>(new TApplication());
+        {
+            try
```

Comment:
Looks good. Please apply the same pattern to the other stores.

Reply:
Thing is this implementation doesn't use `await` so we'd get a compiler warning and pay for a state machine that wouldn't be useful. Let's keep it as-is.